google/cloud/aiplatform/metadata/metadata.py

```diff
@@ -94,11 +119,68 @@ def log_metrics(self, metrics: Dict[str, Union[str, float, int]]):
         )
         artifact.update(metadata=metrics)
 
-    def get_experiment(self, experiment: str):
-        raise NotImplementedError("get_experiment not implemented")
+    def get_experiment(self, experiment: str) -> pd.DataFrame:
```

Comment:
I think experiment should be Optional so if I call `get_experiment()` then I get the experiment dataframe for the current set experiment.

Reply:
Also, I wonder if this should be `get_experiment_dataframe`.

Reply:
Good point, made experiment as optional. 

For the function naming, let me loop in Nate as we had the get_experiment name in the DD. I can get another PR to update the name if we decide to change it.